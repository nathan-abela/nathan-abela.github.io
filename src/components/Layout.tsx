import React from 'react';

import Head from 'next/head';

import styles from '@styles/Home.module.css';

import Container from './Container';
import Footer from './Footer';
import Nav from './Nav';

type LayoutProps = {
    children?: React.ReactNode;
    title?: string;
};

const Layout: React.FC<LayoutProps> = ({ title = 'Nathan Abela | Portfolio', children }) => (
    <div className={styles.container}>
        <Head>
            <title>{title}</title>
            <link rel='icon' type='image/x-icon' href='/favicon.ico' />
            <meta charSet='utf-8' />
            <meta name='author' content='Nathan Abela' />
            <meta name='viewport' content='width=device-width, initial-scale=1.0' />
            <meta property='og:title' content='Nathan Abela | Portfolio' />
            <meta property='og:type' content='website' />
        </Head>
        <Nav />
        <Container justifyContent='space-between' alignContent='space-between'>
            <main className={styles.main}>{children}</main>
            <Footer />
        </Container>
    </div>
);

export default Layout;
