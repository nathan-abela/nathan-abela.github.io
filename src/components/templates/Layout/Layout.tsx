import { FC, ReactNode } from 'react';
import { Footer } from 'Organisms/Footer';
import { Header } from 'Organisms/Header';
import { StyledLayout } from './styles';

interface LayoutProps {
  children: ReactNode;
}

const Layout: FC<LayoutProps> = ({ children }) => {
  return (
    <StyledLayout>
      <Header />
      <main>{children}</main>
      <Footer />
    </StyledLayout>
  );
};

export { Layout };
