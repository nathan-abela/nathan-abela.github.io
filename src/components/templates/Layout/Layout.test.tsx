import { render } from '@testing-library/react';

import { Layout } from './Layout';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
	useSelector: jest.fn(),
	useDispatch: () => mockDispatch,
}));

describe('Layout', () => {
	it('renders correctly', () => {
		const { container } = render(
			// eslint-disable-next-line prettier/prettier
			<Layout>Test Content</Layout>
		);
		expect(container).toMatchSnapshot();
	});
});
