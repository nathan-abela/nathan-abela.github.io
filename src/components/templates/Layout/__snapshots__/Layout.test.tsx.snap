// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div>
  <div
    class="styles__StyledLayout-sc-10u3re6-0 gmhIHB"
  >
    <header>
      <div
        class="styles__StyledContainer-sc-a6yh8x-0 earCTD"
      >
        <div
          class="styles__Flex-sc-1w2me4-0 bHDEaA"
        >
          <h1>
            <a
              href="/"
            >
              Nathan Abela
            </a>
          </h1>
          <div
            class="styles__Desktop-sc-1w2me4-1 kRNKOm"
          >
            <nav
              class="styles__StyledNavigation-sc-1f4kyre-0"
            >
              <ul
                class="styles__List-sc-1f4kyre-1 ivSflr"
              >
                <li>
                  <a
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li>
                  <a
                    href="/about"
                  >
                    About
                  </a>
                </li>
                <li>
                  <a
                    href="/education"
                  >
                    Education
                  </a>
                </li>
                <li>
                  <a
                    href="/work"
                  >
                    Work
                  </a>
                </li>
                <li>
                  <a
                    href="/portfolio"
                  >
                    Portfolio
                  </a>
                </li>
                <li
                  class="desktop-only"
                >
                  <button
                    class="styles__StyledThemeToggle-sc-1b7mai8-0 kHwexB"
                    title="Switch to light theme"
                  >
                    <test-file-stub />
                  </button>
                </li>
              </ul>
            </nav>
          </div>
          <div
            class="styles__Mobile-sc-1w2me4-2 kpxcyD"
          >
            <ul
              class="styles__StyledMobileNavigation-sc-1619527-0 huyxbx"
            >
              <li>
                <button
                  class="styles__StyledThemeToggle-sc-1b7mai8-0 kHwexB"
                  title="Switch to light theme"
                >
                  <test-file-stub />
                </button>
              </li>
              <li>
                <button
                  class="styles__Menu-sc-1619527-1 hKtafq"
                >
                  <test-file-stub />
                </button>
              </li>
            </ul>
            <div
              class="styles__Overlay-sc-1619527-2 lnRURg"
            >
              <nav
                class="styles__StyledNavigation-sc-1f4kyre-0"
              >
                <ul
                  class="styles__List-sc-1f4kyre-1 ivSflr"
                >
                  <li>
                    <a
                      href="/"
                    >
                      Home
                    </a>
                  </li>
                  <li>
                    <a
                      href="/about"
                    >
                      About
                    </a>
                  </li>
                  <li>
                    <a
                      href="/education"
                    >
                      Education
                    </a>
                  </li>
                  <li>
                    <a
                      href="/work"
                    >
                      Work
                    </a>
                  </li>
                  <li>
                    <a
                      href="/portfolio"
                    >
                      Portfolio
                    </a>
                  </li>
                  <li
                    class="desktop-only"
                  >
                    <button
                      class="styles__StyledThemeToggle-sc-1b7mai8-0 kHwexB"
                      title="Switch to light theme"
                    >
                      <test-file-stub />
                    </button>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </div>
      </div>
    </header>
    <main>
      Test Content
    </main>
    <footer
      class="styles__StyledFooter-sc-1pzkd2j-0 bbUQND"
    >
      <div
        class="styles__StyledContainer-sc-a6yh8x-0 earCTD"
      >
        <div
          class="styles__Flex-sc-1pzkd2j-1 dfBZPt"
        >
          <span>
            Â© 
            2025
             Nathan Abela
          </span>
          <ul
            class="styles__StyledSocialIcons-sc-1vfff3-0 fBtYTT"
          >
            <li>
              <a
                href="https://github.com/nathan-abela"
                rel="noopener noreferrer"
                target="_blank"
                title="Github"
              >
                <test-file-stub />
              </a>
            </li>
            <li>
              <a
                href="https://www.linkedin.com/in/nathan-abela"
                rel="noopener noreferrer"
                target="_blank"
                title="LinkedIn"
              >
                <test-file-stub />
              </a>
            </li>
            <li>
              <a
                href="https://www.youtube.com/@nathanabela"
                rel="noopener noreferrer"
                target="_blank"
                title="YouTube"
              >
                <test-file-stub />
              </a>
            </li>
            <li>
              <a
                href="mailto:nathanabela7@gmail.com"
                rel="noopener noreferrer"
                target="_blank"
                title="Email"
              >
                <test-file-stub />
              </a>
            </li>
          </ul>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
